### PR TITLE
Types refactor

### PR DESCRIPTION
**Pull request by Tryptophan**
Marking as do not merge (DNM) but will make incremental changes to westegg for the types refactor here and merge in once complete.

### CODE REVIEW

**Comment by cjayross** [comment]
You don't need to comment out all of your existing types. You can instead just remove them. They will still be visible in the git history.

**Comment by Tryptophan** [comment]
@cjayross this should be good to merge whenever you think it looks good, look over the changes and let me know if anything should change. I'll need some updates to types for GET requests, but this is most of the old functionality for uploading videos.

File: src/middleware/handled.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import validator from "validator";
-import { Channel } from "../entities/Channel";
 import { User } from "../entities/User";
-import { ErrorCode, WestEggError } from "../errors";
+import { ErrorCode, WestEggError } from "@gatsby-tv/types";
+import { Channel } from "../entities/Channel";
 
 const HANDLE_MIN_LENGTH = 3;
```
Comment (cjayross) [inline src/middleware/handled.ts:6]: Can you make the minimum handle length 4? This way they don't conflict with the `api` route I have on the frontend. This is also the same length that both Twitch and Twitter requires as well.

File: src/middleware/named.ts [hunk 2]
```diff
@@ -1,12 +1,12 @@
-import { ErrorCode, WestEggError } from "../errors";
+import { ErrorCode, WestEggError } from "@gatsby-tv/types";
 
 const DISPLAY_NAME_MIN_LENGTH = 3;
```
Comment (cjayross) [inline src/middleware/named.ts:3]: Display names can be whatever the hell people want for the most part. The maximum length should be 50 characters and must not be an empty string.

**Review by cjayross — COMMENTED** [review]
So I looked it over, we won't be needing anything to do with passwords since I'll handle that on my end. As such, I will need `/login`  to allow me to submit a `UserId` string that you can look up in the database. If it fails, then I know that I need to request additional account information like their handle and display name. The `/signup` endpoint will be were I send that additional info so that future login attempts succeed and subsequently return a web token.

File: src/middleware/named.ts (hunk 2 above)
Comment (Tryptophan) [reply src/middleware/named.ts:3]: So minimum of 1 character?

**Comment by Tryptophan** [comment]
> So I looked it over, we won't be needing anything to do with passwords since I'll handle that on my end. As such, I will need `/login` to allow me to submit a `UserId` string that you can look up in the database. If it fails, then I know that I need to request additional account information like their handle and display name. The `/signup` endpoint will be were I send that additional info so that future login attempts succeed and subsequently return a web token.

Can you update the types repo to reflect that for the Signup/LoginRequest types? that way I can build out the requests.

File: src/entities/Base.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { IPFSContent } from "@gatsby-tv/types";
+import mongoose, { Schema } from "mongoose";
+
+// TODO: Get IPFS hash to default profile pic
+export const DEFAULT_AVATAR: IPFSContent = {
+  hash: "1234",
+  mimeType: "JPG"
```
[the remark this answers is not in the document]
Comment (Tryptophan) [reply src/entities/Base.ts]: fixed

File: src/middleware/named.ts (hunk 2 above)
Comment (Tryptophan) [reply src/middleware/named.ts]: fixed

**Comment by cjayross** [comment]
That sounds perfectly reasonable. I've been working on the frontend mostly as of late, though technically a rework for a number of different types is currently being worked on in the `frontend_types` branch.

**Pull request merged**